### DISCUSSION TITLE
Missing values in Testcase 5 output files

### DISCUSSION THREAD

**Discussion by tais958** [NASA-LIS/LISF#1063] [Q&A] [answered]
Hello, 

I tried to run through `testcase 5`, and it all went smoothly and the log file seems to be normal as it shows below.
 
 Writing to LDT output directory: ./
 Writing CDF file .//./cdf_smapobs.nc
 Writing CDF domain file .//./cdf_smapobs_domain.nc
 Successfully wrote CDF file .//./cdf_smapobs.nc
 Successfully wrote CDF file .//./cdf_smapobs_domain.nc
 --------------------------------
  Finished LDT run
 --------------------------------

However, the `cdf_smapobs.nc` and `cdf_smapobs_domain.nc` do not look right as the variables are all filled with missing values. I've checked the SMAP obs data files and others, they seem to be read in fine. Does anyone happen to know how this could happen? Could it be due to the HDF5 is not used when compiling?  

Thanks!

**Comment by bmcandr** [comment, marked as answer]
I strongly suspect this is somewhat related to #1059. Just as step 1 requires HDF4, step 5 requires HDF5 to read the SMAP data (HDF5 files).

Unfortunately, our preprocessor checks for HDF5 support do not write an error message to the log if this requirement is not satisfied--it simply skips over the code contained within the [`#if (defined USE_HDF5)`](https://github.com/NASA-LIS/LISF/blob/0e4b2ec81d1a2fa93d44d1a5efd910bf2e188630/ldt/DAobs/NASA_SMAPsm/readNASASMAPsmObs.F90#L261) block.

I'm also noticing that in this case the ldtlog file will inaccurately indicate that SMAP files are being read because [the log message is written _before_ the calls to `read_NASASMAP_data` and `read_SMAPL2sm_data`](https://github.com/NASA-LIS/LISF/blob/0e4b2ec81d1a2fa93d44d1a5efd910bf2e188630/ldt/DAobs/NASA_SMAPsm/readNASASMAPsmObs.F90#L116), the majority of which are skipped if LDT has not been built with HDF5 support.

Rebuilding with HDF5 support is the only way forward with this particular reader. I will open an issue about adding log messages that notify users about the HDF5 requirement. Thanks again for bringing these issues to our attention.

**Reply by bmcandr** [reply]
Just opened #1064 to address this.

**Reply by bmcandr** [reply]
And FYI: step 6 will require that LIS is built with HDF5 as the same SMAP files will be read in this step...!

**Reply by tais958** [reply]
@bmcandr, thanks for the heads up! I am trying to recompile them with HDF5. I think, in this case, it would be useful to warn users that it's strongly recommended the LISF should be compiled with HDF4 as well as HDF5. Learned some from the process though...

**Reply by tais958** [reply]
@bmcandr Just got error when building LDT with HDF5 as below:

mpiifort -o LDT ANSASNWDsnow_obsMod.o ASCATTUWsm_obsMod.o AquariusL2sm_obsMod.o BayesianMergingMod.o CLM2_parmsMod.o CLM45_parmsMod.o CLSMF25_parmsMod.o Crocus_parmsMod.o ESACCIsm_obsMod.o FLAKE_parmsMod.o GCOMWAMSR2TB_ANNdataMod.o GCOMW_AMSR2L3sm_obsMod.o GCOMW_AMSR2L3snd_obsMod.o GHCN_ANNdataMod.o GLASSlai_obsMod.o GRACEQLtws_obsMod.o GRACEtws_obsMod.o GeoWRSI_parmsMod.o HYMAP_parmsMod.o JULES50_parmsMod.o LDT_ANNMod.o LDT_ANNdata_FTable.o LDT_ANNdata_pluginMod.o LDT_CDFMod.o LDT_DAmetricsDataMod.o LDT_DAmetricsMod.o LDT_DAobsDataMod.o LDT_DAobs_FTable.o LDT_DAobs_pluginMod.o LDT_DAobservationsMod.o LDT_DApreprocMod.o LDT_DrangeMod.o LDT_FORC_AttributesMod.o LDT_LMLCMod.o LDT_LSMCropModifier_Mod.o LDT_LSMparamProcMod.o LDT_LSMparams_FTable.o LDT_MuMod.o LDT_NUWRFprocMod.o LDT_OPTUEMod.o LDT_PRIV_gridMod.o LDT_PRIV_rcMod.o LDT_PRIV_tileMod.o LDT_SigmaMod.o LDT_SimpleWeight_TdscaleMod.o LDT_SurfaceTypeMod.o LDT_albedoMod.o LDT_albedo_FTable.o LDT_binaryIOlayer_module.o LDT_bratsethMod.o LDT_climateParmsMod.o LDT_climoRstProcMod.o LDT_climparms_FTable.o LDT_constantsMod.o LDT_coreMod.o LDT_create_subdirs.o LDT_domainMod.o LDT_domain_FTable.o LDT_domain_pluginMod.o LDT_ensRstMod.o LDT_fileIOMod.o LDT_final_MetforcProc.o LDT_forcingparms_Ftable.o LDT_gfracMod.o LDT_gfrac_FTable.o LDT_glacierFractionMod.o LDT_glacierMod.o LDT_glacier_FTable.o LDT_gridmappingMod.o LDT_historyMod.o LDT_init_ANNproc.o LDT_init_DApreproc.o LDT_init_EnsRstpreproc.o LDT_init_LISHydropreproc.o LDT_init_LSMparamproc.o LDT_init_MetTimeDScale.o LDT_init_MetforcProc.o LDT_init_NUWRFpreproc.o LDT_init_OPTUEparamproc.o LDT_init_StatDscaleMetForc.o LDT_init_climoRstproc.o LDT_init_rstTransformProc.o LDT_init_usafsi.o LDT_irrig_Ftable.o LDT_irrigationMod.o LDT_laisaiMod.o LDT_laisai_FTable.o LDT_lakeParamProcMod.o LDT_lakeparams_FTable.o LDT_landcover_FTable.o LDT_logMod.o LDT_mapSurfaceModelType.o LDT_mask_FTable.o LDT_metforcDscaleMod.o LDT_metforcScale_pluginMod.o LDT_metforcdscale_FTable.o LDT_metforcingMod.o LDT_metforcingParmsMod.o LDT_metforcing_FTable.o LDT_metforcing_pluginMod.o LDT_mpDecomp.o LDT_mpiMod.o LDT_noahparms_FTable.o LDT_numericalMethodsMod.o LDT_openwaterMod.o LDT_paramDataMod.o LDT_paramMaskCheckMod.o LDT_paramOptCheckMod.o LDT_paramProcMod.o LDT_paramTileInputMod.o LDT_param_pluginMod.o LDT_pluginIndices.o LDT_ran2_gasdev.o LDT_readConfig.o LDT_routingParamProcMod.o LDT_routingparams_FTable.o LDT_rstTransformProcMod.o LDT_run_ANNproc.o LDT_run_DApreproc.o LDT_run_EnsRstpreproc.o LDT_run_LISHydropreproc.o LDT_run_LSMparamproc.o LDT_run_MetTimeDScale.o LDT_run_MetforcProc.o LDT_run_NUWRFpreproc.o LDT_run_OPTUEparamproc.o LDT_run_StatDscaleMetForc.o LDT_run_climoRstproc.o LDT_run_rstTransformProc.o LDT_run_usafsi.o LDT_runmode_FTable.o LDT_runmode_pluginMod.o LDT_soilsMod.o LDT_soils_FTable.o LDT_spatialDownscalingMod.o LDT_statDscaleMod.o LDT_statDscale_pluginMod.o LDT_statdscale_FTable.o LDT_timeMgrMod.o LDT_topoMod.o LDT_topo_FTable.o LDT_usafsiMod.o LDT_vegdataMod.o LDT_vic412rstMod.o LDT_xmrg_reader.o LDTmain.o LISlsmSM_ANNdataMod.o LISlsmSM_obsMod.o LPRM_AMSREsm_ANNdataMod.o LPRM_AMSREsm_obsMod.o LPRMvod_obsMod.o MCD15A2Hlai_obsMod.o MOD10A1_ANNdataMod.o MODISlst_ANNdataMod.o Monfredaetal08_crops_module.o Mosaic_parmsMod.o NASASMAPsm_obsMod.o NASASMAPvod_obsMod.o NASA_AMSREsm_obsMod.o Noah_parmsMod.o RDHM_parmsMod.o RFE2Daily_forcingMod.o RFE2gdas_forcingMod.o RUC_parmsMod.o SACHTET_parmsMod.o SMMRSNWDsnow_obsMod.o SMOPSsm_obsMod.o SMOSL2sm_obsMod.o SMOSNESDISsm_obsMod.o SMOSNRTNNL2sm_obsMod.o SSMISNWDsnow_obsMod.o SiB2_parmsMod.o Snow17_parmsMod.o TRMM3B42RTV7_forcingMod.o TRMM3B42V6_forcingMod.o TRMM3B42V7_forcingMod.o USAFSI_amsr2Mod.o USAFSI_analysisMod.o USAFSI_arraysMod.o USAFSI_binary.o USAFSI_galwemMod.o USAFSI_gofsMod.o USAFSI_lisMod.o USAFSI_netcdfMod.o USAFSI_paramsMod.o USAFSI_run.o USAFSI_ssmisMod.o USAFSI_utilMod.o USAFSI_xcalgmiMod.o UTM_utils.o VIC_parmsMod.o WindSatsm_obsMod.o agrradps_forcingMod.o apply_conusmask.o assigncroptype.o bilinear_interp.o bilinear_interp_input.o bilinear_interp_input_withgrid.o breakpt_module.o calc_SlopeAspect_module.o chirps2_forcingMod.o cm_readzipf.o cmap_forcingMod.o cmorph_forcingMod.o compute_earth_coord.o compute_earth_coord_UTM.o compute_earth_coord_ease.o compute_earth_coord_gauss.o compute_earth_coord_hrap.o compute_earth_coord_lambert.o compute_earth_coord_latlon.o compute_earth_coord_latlon_ll.o compute_earth_coord_latlon_ur.o compute_earth_coord_ll.o compute_earth_coord_merc.o compute_earth_coord_polar.o compute_earth_coord_ur.o compute_grid_coord.o compute_grid_coord_ease.o compute_grid_coord_gauss.o compute_grid_coord_hrap.o compute_grid_coord_lambert.o compute_grid_coord_latlon.o compute_grid_coord_merc.o compute_grid_coord_polar.o compute_stnwts.o compute_vinterp_weights.o conserv_interp.o conserv_interp_input.o conserv_interp_input_withgrid.o create_SACsoilparms_Korenv1.o create_gdasf9_filename.o create_gdasfilename.o create_gfs_f0backup_filename.o create_gfsfilename.o create_maskfile.o create_nam242f9_filename.o create_nam242filename.o create_soil_albedo_from_whs.o create_soil_params_from_cap.o diff_lon.o easeV2_utils.o ecmwf_forcingMod.o ecmwfreanal_forcingMod.o era5_forcingMod.o ezlh_convert.o ezlh_inverse.o fbil_module.o finalize_RFE2Daily.o finalize_RFE2gdas.o finalize_TRMM3B42RTV7.o finalize_TRMM3B42V6.o finalize_TRMM3B42V7.o finalize_chirps2.o finalize_cmap.o finalize_cmorph.o finalize_ecmwf.o finalize_ecmwfreanal.o finalize_era5.o finalize_gdas.o finalize_geos5fcst.o finalize_gfs.o finalize_merra2.o finalize_merraland.o finalize_nam242.o finalize_nldas1.o finalize_nldas2.o finalize_princeton.o finalize_stg2.o finalize_stg4.o fmktime.o forcingClimoMod.o gaussian_quadrature.o gaussian_routines.o gausslat.o gdas3d_forcingMod.o gdas_forcingMod.o geogfill2_ecmwfreanal.o geos5fcst_forcingMod.o geowrsi2_arraymgmt_module.o geowrsi2_set_gcoords.o get_RFE2Daily.o get_RFE2gdas.o get_TRMM3B42RTV7.o get_TRMM3B42V6.o get_TRMM3B42V7.o get_chirps2.o get_cmap.o get_cmorph.o get_ecmwf.o get_ecmwfreanal.o get_era5.o get_fieldpos.o get_gdas.o get_geos5fcst.o get_gesdisc_filenames.o get_gfs.o get_merra2.o get_merraland.o get_nam242.o get_ncep_filenames.o get_nldas1.o get_nldas1_filename.o get_nldas2.o get_princeton.o get_stg2.o get_stg4.o gfs_forcingMod.o gldas_forcingMod.o gswp1_forcingMod.o gswp2_forcingMod.o hrapToLatLon.o idw_module.o interp_TRMM3B42RTV7.o interp_TRMM3B42V6.o interp_TRMM3B42V7.o interp_cmap.o interp_cmorph.o interp_geos5fcst.o interp_nldas1.o interp_stg2.o interp_stg4.o interp_stndata.o latlonTohrap.o latlonTopolar.o lltops.o map_utils.o merra2_forcingMod.o merraland_forcingMod.o mqb_module.o nam242_forcingMod.o narr_forcingMod.o neighbor_interp.o neighbor_interp_input.o neighbor_interp_input_withgrid.o nldas1_ec_removal.o nldas1_forcingMod.o nldas2_ec_removal.o nldas2_forcingMod.o normalize_stnwts.o polarToLatLon.o polfixs.o princeton_forcingMod.o pstoll.o rdhm356_file_name.o rdhm356_finalize.o rdhm356_forcingMod.o rdhm356_get_forcing.o rdhm356_interp_precip.o rdhm356_interp_temper.o rdhm356_read_precip.o rdhm356_read_temper.o rdhm356_readcrd.o readANSASNWDsnowObs.o readASCATTUWsmObs.o readAquariusL2smObs.o readESACCIsmObs.o readGCOMWAMSR2TbANNdata.o readGCOMW_AMSR2L3smObs.o readGCOMW_AMSR2L3sndObs.o readGHCNANNdata.o readGLASSlaiObs.o readGRACEQLtwsObs.o readGRACEtwsObs.o readLISlsmSMANNdata.o readLISlsmSMObs.o readLPRM_AMSRE_ANNdata.o readLPRM_AMSREsmObs.o readLPRMvodObs.o readMCD15A2HlaiObs.o readMOD10A1ANNdata.o readMODISlstANNdata.o readNASASMAPsmObs.o readNASASMAPvodObs.o readNASA_AMSREsmObs.o readRFE2gdascrd.o readSMMRSNWDsnowObs.o readSMOPSsmObs.o readSMOSL2smObs.o readSMOSNESDISsmObs.o readSMOSNRTNNL2smObs.o readSSMISNWDsnowObs.o readWindSatsmObs.o read_ALMIPII_droot.o read_ALMIPII_dsoil.o read_ALMIPII_lc.o read_ALMIPII_soilfractions.o read_AVHRRNative_gfrac.o read_AVHRR_GFS_lc.o read_AVHRR_gfrac.o read_AVHRR_lai.o read_AVHRR_lc.o read_AVHRR_sai.o read_BarlageNative_maxsnoalb.o read_BrieglebNative_monalbedo.o read_BrieglebNative_qtralbedo.o read_Briegleb_albedo.o read_CLM45_lc.o read_CLSMF25_albnir.o read_CLSMF25_albvis.o read_CLSMF25_araparams.o read_CLSMF25_arsparams.o read_CLSMF25_arwparams.o read_CLSMF25_bedrockdepth.o read_CLSMF25_bexp.o read_CLSMF25_bfparams.o read_CLSMF25_gfrac.o read_CLSMF25_gfracmax.o read_CLSMF25_gfracmin.o read_CLSMF25_gnuparam.o read_CLSMF25_ksat.o read_CLSMF25_lai.o read_CLSMF25_laimax.o read_CLSMF25_laimin.o read_CLSMF25_lc.o read_CLSMF25_porosity.o read_CLSMF25_psisat.o read_CLSMF25_tauparams.o read_CLSMF25_tsparams.o read_CLSMF25_wpwet.o read_CONSTANT_albedo.o read_CONSTANT_aspect.o read_CONSTANT_elev.o read_CONSTANT_gfrac.o read_CONSTANT_gfracmax.o read_CONSTANT_gfracmin.o read_CONSTANT_lai.o read_CONSTANT_lc.o read_CONSTANT_mxsnoalb.o read_CONSTANT_porosity.o read_CONSTANT_sai.o read_CONSTANT_slope.o read_CONSTANT_slopetype.o read_CONSTANT_slopetype_cro.o read_CONSTANT_soilfractions.o read_CONSTANT_tbot.o read_CONSTANT_tbot_cro.o read_CONSTANT_texture.o read_FAO_color.o read_FAO_porosity.o read_FAO_soilfractions.o read_FAO_texture.o read_FLake_lakedepth.o read_GFS_maxsnoalb.o read_GFS_slopetype.o read_GLIMS_glacierfraction.o read_GLIMS_glaciermask.o read_GLWD_inlandwatertype.o read_GRIPC_irrigfrac.o read_GRIPC_irrigtype.o read_GTOPO30Native_elev.o read_GTOPO30_GFS_elev.o read_GTOPO30_aspect.o read_GTOPO30_curv.o read_GTOPO30_elev.o read_GTOPO30_slope.o read_GeoWRSI2_landmask.o read_GeoWRSI2_lgp.o read_GeoWRSI2_sos.o read_GeoWRSI2_sosanom.o read_GeoWRSI2_sosclim.o read_GeoWRSI2_whc.o read_GeoWRSI2_wrsiclim.o read_HYMAP_baseflow_delay.o read_HYMAP_baseflow_dwi_ratio.o read_HYMAP_basin.o read_HYMAP_basin_mask.o read_HYMAP_drain_area.o read_HYMAP_fld_height.o read_HYMAP_fld_z.o read_HYMAP_flow_dir_x.o read_HYMAP_flow_dir_y.o read_HYMAP_flow_type.o read_HYMAP_grid_area.o read_HYMAP_grid_dist.o read_HYMAP_grid_elev.o read_HYMAP_refq.o read_HYMAP_river_height.o read_HYMAP_river_length.o read_HYMAP_river_width.o read_HYMAP_river_z.o read_HYMAP_runoff_delay.o read_HYMAP_runoff_delaym.o read_HYMAP_runoff_dwi_ratio.o read_ISA_lc.o read_ISA_maskfile.o read_ISLSCP1_cro_tbot.o read_ISLSCP1_tbot.o read_ISRIC_soilfractions.o read_ISRIC_texture.o read_MERIT1K_aspect.o read_MERIT1K_elev.o read_MERIT1K_slope.o read_MODISNative_PFT.o read_MODISNative_lc.o read_MODIS_lc.o read_NCEPNative_shdmax.o read_NCEPNative_shdmin.o read_NCEPNative_slopetype.o read_NCEP_GFS_cro_tbot.o read_NCEP_GFS_tbot.o read_NCEP_cro_tbot.o read_NCEP_shdmax.o read_NCEP_shdmin.o read_NCEP_slopetype.o read_NCEP_tbot.o read_NLDAS_climppt.o read_OzdoganGutman_irrigfrac.o read_PRISM_climppt.o read_RUC_ISLSCP1tbot.o read_RUC_NCEPLIS_slopetype.o read_RUC_NCEPLIS_tbot.o read_RUC_NCEPslopetype.o read_RobinsonKuklaNative_mxsnoalb.o read_RobinsonKukla_mxsnoalb.o read_SACHTET356_gfrac.o read_SACHTET356_lc.o read_SACHTET356_mxsnoalb.o read_SACHTET356_pet.o read_SACHTET356_petadj.o read_SACHTET356_tbot.o read_SRTM_Native_aspect.o read_SRTM_Native_elev.o read_SRTM_Native_slope.o read_SRTM_aspect.o read_SRTM_elev.o read_SRTM_slope.o read_STATSGOFAO_Native_texture.o read_STATSGOFAO_texture.o read_STATSGOv1_hydrosoilgroup.o read_STATSGOv1_soilfractions.o read_STATSGOv1_texture.o read_SiB2_parms.o read_Special_soilfractions.o read_Special_texture.o read_TRMM3B42RTV7.o read_TRMM3B42V6.o read_TRMM3B42V7.o read_UKMO_IGBP_PFT.o read_UMDCROPMAP_croptype.o read_UM_ancillary.o read_USGSNative_lc.o read_USGS_lc.o read_UserDerived_irrigfrac.o read_VIC411_lc.o read_VIC412_lc.o read_WorldClim_climppt.o read_ZOBLER_GFS_texture.o read_chirps2.o read_cmap.o read_cmorph.o read_ecmwf.o read_ecmwf_elev.o read_ecmwfelev_ldtproc.o read_ecmwfreanal.o read_ecmwfreanal_elev.o read_era5.o read_era5_elev.o read_gdas.o read_gdas_elev.o read_gdaselev_ldtproc.o read_geos5fcst.o read_gfs.o read_gfs_elev.o read_maskfile.o read_maskfile_UKMO_CAP_Netcdf.o read_maskfile_UKMO_IGBP_Native_PFT.o read_merra2.o read_merra2_elev.o read_merraland.o read_nam242.o read_nam242_elev.o read_netcdf_4d.o read_nldas1.o read_nldas1_elev.o read_nldas2_elev.o read_nldas2a.o read_nldas2b.o read_princeton.o read_princeton_elev.o read_regmask_gis.o read_regmask_mask.o read_regmask_wrsi.o read_stg2.o read_stg4.o read_um_soil_params.o readconfig_chirps2.o readcrd_RFE2Daily.o readcrd_TRMM3B42RTV7.o readcrd_TRMM3B42V6.o readcrd_TRMM3B42V7.o readcrd_cmap.o readcrd_cmorph.o readcrd_ecmwf.o readcrd_ecmwfreanal.o readcrd_era5.o readcrd_gdas.o readcrd_geos5fcst.o readcrd_gfs.o readcrd_merra2.o readcrd_merraland.o readcrd_nam242.o readcrd_nldas1.o readcrd_nldas2.o readcrd_princeton.o readcrd_stg2.o readcrd_stg4.o readcropinventory.o readinput_easev2.o readinput_gaussian.o readinput_hrap.o readinput_lambert.o readinput_latlon.o readinput_merc.o readinput_polar.o readmask_ecmwfreanal.o readprecip_RFE2Daily.o readprecip_RFE2gdas.o readsimGRACEJPLObs.o readsyntheticsmANNdata.o readsyntheticsmObs.o readzipf.o reproject_RFE2Daily.o reproject_RFE2gdas.o reset_RFE2Daily.o reset_RFE2gdas.o reset_chirps2.o reset_cmap.o reset_ecmwf.o reset_era5.o reset_gdas.o reset_geos5fcst.o reset_merra2.o reset_merraland.o reset_nam242.o reset_nldas1.o reset_nldas2.o reset_princeton.o reset_stg4.o setAlbedoParmsFullnames.o setClimateParmsFullnames.o setCropParmsFullnames.o setGfracParmsFullnames.o setGlacierFracParmsFullnames.o setIrrigParmsFullnames.o setLAISAIParmsFullnames.o setTopoParmsFullnames.o set_AVHRR_lai_attribs.o set_CLSMF25_gfrac_attribs.o set_CLSMF25_lai_attribs.o set_CONSTANT_texture_attribs.o set_ISRIC_texture_attribs.o set_STATSGOFAO_LIS_texture_attribs.o set_STATSGOFAO_Native_texture_attribs.o set_STATSGOv1_hsg_attribs.o set_STATSGOv1_texture_attribs.o set_Special_texture_attribs.o set_ZOBLER_GFS_texture_attribs.o set_albedo_attribs.o set_gfrac_attribs.o set_glacierfraction_attribs.o set_irrigation_attribs.o simGRACEJPL_obsMod.o stg2_forcingMod.o stg2file.o stg4_forcingMod.o stg4file.o stninterp_module.o syntheticsm_ANNdataMod.o syntheticsm_obsMod.o timeinterp_RFE2Daily.o timeinterp_RFE2gdas.o timeinterp_TRMM3B42RTV7.o timeinterp_TRMM3B42V6.o timeinterp_TRMM3B42V7.o timeinterp_chirps2.o timeinterp_cmap.o timeinterp_cmorph.o timeinterp_ecmwf.o timeinterp_ecmwfreanal.o timeinterp_era5.o timeinterp_gdas.o timeinterp_geos5fcst.o timeinterp_gfs.o timeinterp_merra2.o timeinterp_merraland.o timeinterp_nam242.o timeinterp_nldas1.o timeinterp_nldas2.o timeinterp_princeton.o timeinterp_rdhm356.o timeinterp_stg2.o timeinterp_stg4.o upscaleByAveraging.o upscaleByAveraging_input.o upscaleByCnt.o upscaleByMode.o vinterp.o write_lookup_table.o xmrg_read_data_c.o xmrg_read_header_c.o xmrg_reverse_byte.o zterp.o ztif.o ztif_frac.o ztif_frac_slice.o ztif_read.o ztif_write.o  -L/qfs/projects/ops/rh6/esmf/7.1/lib/libO/Linux.intel.64.intelmpi.default -lesmf -lstdc++ -limf -lm -lrt -lz -L/pic/projects/sooty2/tais958/lib/eccodes-2.12.5_intel-16.0.2/lib/ -leccodes_f90 -leccodes -L/share/apps/jasper/1.900.1//lib64/ -ljasper -L/share/apps/netcdf/4.3.2/intel/16.0.2/lib/ -lnetcdff -lnetcdf -L/pic/projects/sooty2/tais958/lib/hdf-eos2-3.0_intel16.0.2/lib/ -lhdfeos -lGctp -L/pic/projects/sooty2/tais958/lib/hdf-4.2.15_intel-16.0.2/lib/ -lmfhdf -ldf -ljpeg -lz  -L/pic/projects/sooty2/tais958/lib/hdf5-1.12.1_intel-16.0.2/lib/ -lhdf5_fortran -lhdf5_hl -lhdf5 -ldl -mkl
readGLASSlaiObs.o: In function `readglasslaiobs_':
readGLASSlaiObs.F90:(.text+0x801): undefined reference to `gdopen_'
readGLASSlaiObs.F90:(.text+0x850): undefined reference to `gdattach_'
readGLASSlaiObs.F90:(.text+0x961): undefined reference to `gddetach_'
readGLASSlaiObs.F90:(.text+0x975): undefined reference to `gdclose_'
readGLASSlaiObs.F90:(.text+0x13ba): undefined reference to `gdrdfld_'
readGLASSlaiObs.F90:(.text+0x17f9): undefined reference to `gdrdfld_'
readGLASSlaiObs.o: In function `read_glasslai_data_':
readGLASSlaiObs.F90:(.text+0x1ef3): undefined reference to `gdopen_'
readGLASSlaiObs.F90:(.text+0x1f3c): undefined reference to `gdattach_'
readGLASSlaiObs.F90:(.text+0x2046): undefined reference to `gddetach_'
readGLASSlaiObs.F90:(.text+0x2054): undefined reference to `gdclose_'
readGLASSlaiObs.F90:(.text+0x2625): undefined reference to `gdrdfld_'
readGLASSlaiObs.F90:(.text+0x2a12): undefined reference to `gdrdfld_'
readNASA_AMSREsmObs.o: In function `read_amsresm_':
readNASA_AMSREsmObs.F90:(.text+0xc16): undefined reference to `gdopen_'
readNASA_AMSREsmObs.F90:(.text+0xd49): undefined reference to `gdattach_'
readNASA_AMSREsmObs.F90:(.text+0xe43): undefined reference to `gdrdfld_'
readNASA_AMSREsmObs.F90:(.text+0x14b5): undefined reference to `gdrdfld_'
readNASA_AMSREsmObs.F90:(.text+0x1aa8): undefined reference to `gdrdfld_'
readNASA_AMSREsmObs.F90:(.text+0x1ebd): undefined reference to `gddetach_'
readNASA_AMSREsmObs.F90:(.text+0x1f5e): undefined reference to `gdclose_'
readNASA_AMSREsmObs.F90:(.text+0x25d6): undefined reference to `gddetach_'
readNASA_AMSREsmObs.F90:(.text+0x25e4): undefined reference to `gdclose_'
readNASA_AMSREsmObs.F90:(.text+0x283f): undefined reference to `gddetach_'
readNASA_AMSREsmObs.F90:(.text+0x284d): undefined reference to `gdclose_'
readNASA_AMSREsmObs.F90:(.text+0x2b25): undefined reference to `gddetach_'
readNASA_AMSREsmObs.F90:(.text+0x2b33): undefined reference to `gdclose_'
readNASA_AMSREsmObs.F90:(.text+0x2cd9): undefined reference to `gdclose_'
make: *** [LDT] Error 1

[ERR] Compile failed
--------------------------------------------------------------------
Realized this could be due to HDF-EOS library. Any advise is much appreciated!

**Reply by tais958** [reply]
<update> 
I was able to successfully run through testcase 5 with LDT built with HDF5 but without HDF-EOS. The error above was absent when the LDT is not built with HDF-EOS.
